tests/rptest/e2e_tests/flink_scale_test.py

```diff
@@ -82,54 +88,140 @@ def assert_failed_jobs(job_list):
             self.kafkacli.create_topic(spec)
+            self.topic_specs.append(spec)
 
         # Start Flink
         self.flink.start()
 
         # Load python workload to target node
+        # Hardcoded file
         # TODO: Add workload config management
+        # Currently, each INSERT operator will generate 1 subjob
+        # So this config will generate 256 / 64 jobs
         _workload_config = {
             "log_level": "DEBUG",
             "brokers": self.redpanda.brokers(),
             "producer_group": "flink_group",
             "consumer_group": "flink_group",
-            "topic_name": self.topic_name,
+            "topic_name": "flink_scale_topic_placeholder",
             "mode": "produce",
-            "count": 65535 * 32
+            "count": total_events // topics_per_node
         }
 
         # Get workload
         workloads = self.workload_manager.get_workloads(
             ['flink', 'transactions', 'scale'])
         # Run produce part
-        all = self._run_workloads(workloads, _workload_config)
-        # Assert failed jobs
-        assert_failed_jobs(all)
-        # Wait to finish
-        self.flink.wait(timeout_sec=1800)
+        all_jobs = []
+        for spec in self.topic_specs:
+            _workload_config['topic_name'] = spec.name
+            self.logger.debug("Submitting job with config: \n"
+                              f"{json.dumps(_workload_config, indent=2)}")
+            all_jobs += self._run_workloads(workloads, _workload_config)
 
-        # Run consume part
-        _workload_config['mode'] = "consume"
-        all = self._run_workloads(workloads, _workload_config)
         # Assert failed jobs
-        assert_failed_jobs(all)
+        assert_failed_jobs(all_jobs)
 
         # Wait to finish
         self.flink.wait(timeout_sec=1800)
 
-        # TODO: Validate according to metrics
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="1"} 20029
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="2"} 11483
-        # vectorized_internal_rpc_latency_sum{method="commit_tx",service="tx_gateway",shard="3"} 21542
+        topic_watermarks = []
+        for spec in self.topic_specs:
+            hwm = get_hwm_for_topic(spec.name)
+            topic_watermarks.append(hwm)
+
+        # Print out in one place:
+        for idx in range(len(topic_watermarks)):
+            self.logger.info(f"Topic: {self.topic_specs[idx].name}, "
+                             f"watermark: {topic_watermarks[idx]}")
+        self.logger.info(
+            f"Total messages/High watermark sum: {sum(topic_watermarks)}")
+
+        # Simple validation according to RP topic watermarks
+        sum_hwm = sum(topic_watermarks)
+        assert sum_hwm >= total_events, \
+            "High watermark is less than targer total events: " \
+            f"{sum_hwm} hwm, {total_events} target total"
```

Comment:
wdyt about having more tighter validation?  I think >= isn't good enough and may cause false positives.

Now that we are using a monotonic numeric source, I think we can ensure that each integer appears #jobs  times? That seems like a much tighter validation?  memory wise I think  for 4 million ints is pretty trivial even if we cache all of them (during validation), but we can probably use more tricks to optimize.

Reply:
The goal here is to generate target transaction count towards RP, see if this generate any bad log lines or not. This check is to make sure that event count is not less than original sequence count. Which says that there was retries in the process. This is why there is >= here.
Content validation for scale task can be done later when proper consume part of this test will be developed. Such validation should be on the workload consumer side to eliminate moving a lot of data between ducktape client and flink node.

Reply:
> The goal here is to generate target transaction count towards RP, see if this generate any bad log lines or not.

Goal is also correctness which may or may not produce bad log lines. Moving a lot of data between nodes is fine I think as long as we buffer much in memory. Anyway its okay to defer this to a later time, I can add it as a separate task.

Reply:
I'll create corresponding task in e2e for the backlog